test/actions/ActionsProviderSpec.scala

```diff
@@ -558,6 +560,93 @@ class ActionsProviderSpec extends ControllerUnitTest
         SessionValues.VALID_TAX_YEARS -> validTaxYears))) shouldBe Redirect(UnauthorisedUserErrorController.show())
     }
 
+    "redirect to CisUser data finished urls" should {
+
+      "return a redirect to same page when finished the Filter Action" in {
+        val viewAudit = aViewCisPeriodAudit.copy(taxYear = taxYearEOY,
+          cisPeriod = aContractorDetailsAndPeriodData.copy(labour = Some(500), materialsCost = Some(250)))
+        val aCisUserDataWithoutDeductionAmount = aCisUserData.copy(cis = aCisCYAModel.copy(periodData = Some(aCYAPeriodData)))
+
+        val viewNrsPayload = ViewCisPeriodPayload(
```

Comment:
You are not testing NRS here. A different test is needed for that. This needs to be removed. Further, we should not create objects like that in the tests. We should be using builders instead. When there is no builder created, it should be created and used.

Reply:
I think it might be better if we pair a bit to address those. Especially in relation to the ActionsProviderSpec.